File: src/#1047/solution.js
var removeDuplicates = function (S) {
  let stack = []

  for (let i = 0; i < S.length; i++) {
    if (stack.length === 0) {
      stack.push(S[i])
      continue
    }
    let last = stack[stack.length - 1]
    if (S[i] === last) {
      stack.pop()
    } else {
      stack.push(S[i])
    }
  }
  return stack.join('')
}

File: src/#1047/solution.go
package solution

func removeDuplicates(S string) string {
	stack := []byte{}

	for i := 0; i < len(S); i++ {
		if len(stack) == 0 {
			stack = append(stack, S[i])
			continue
		}
		last := stack[len(stack)-1]
		if last == S[i] {
			stack = stack[:len(stack)-1]
		} else {
			stack = append(stack, S[i])
		}
	}

	return string(stack)

}
